src/pages/Home.jsx

```diff
@@ -1,9 +1,134 @@
+    }
+
+    return array;
+  }
+  // Fetch products data from an API when the component mounts.
+  useEffect(() => {
+    const fetchProducts = async () => {
+      //Call the api.get method to retrieve products.
+      try {
+        const response = await api.get('/products');
+        const products = response.data.products;
+        // Swaps elements within the array to produce a new, randomized order and returns the shuffled array.
+        const shuffledProducts = shuffleArray(products);
+        // Update the newArrivals state with the shuffled array of products.
+        setNewArrivals(shuffledProducts);
+        // Fetch only products with isFeatured is true by filtering the products.
+        setFeaturedProducts(products.filter((product) => product.isFeatured));
+      } catch (error) {
+        // If there is an error during the registration process, catches the error and displays an error message.
+        toast.error('Error fetching products', error);
+      }
+    };
+
+    fetchProducts();
+  }, []);
+
   return (
     <Layout>
-      <p>This is the Home page</p>
+      <div className='w-full mb-10'>
+        <img src={homeImage} alt='HomeImage' className='w-full object-cover h-80' />
+        {/*New Arrivals Section*/}
+        <div className='flex flex-col'>
+          <h1 className='text-left font-bold text-3xl mt-10 mb-4 text-primary'>New Arrivals</h1>
+          <div className='carousel mt-4 overflow-hidden'>
+            <motion.div
+              className='flex space-x-2'
+              drag='x'
+              dragConstraints={{ left: -1000, right: 0 }}
+            >
+              {/* Iterates over each item in the newArrivals array. */}
+              {newArrivals.map((product, index) => (
+                <Card
+                  shadow='sm'
+                  key={index}
+                  isPressable
+                  className='border-2 w-[300px] h-[450px] flex-shrink-0'
+                >
+                  <CardBody className='flext justify-center overflow-visible'>
+                    <Image
+                      radius='lg'
+                      width='100%'
+                      alt={product.name}
+                      className='w-full object-contain'
+                      src={product.imageUrl}
+                    />
+                  </CardBody>
+                  <CardFooter className='text-small justify-around'>
+                    <b>{product.name}</b>
+                    <p className='text-default-500'>${product.price}</p>
+                  </CardFooter>
+                </Card>
+              ))}
+            </motion.div>
+          </div>
+        </div>
+
+        {/* Featured Collection Section */}
+        <div className='flex flex-col mt-10'>
+          <h1 className='text-left font-bold text-3xl mb-4 mt-5 text-primary '>
+            Featured Collection
+          </h1>
+          <div className='carousel mt-4 overflow-hidden'>
+            <motion.div
```

Comment:
We also need to move this `motion.div` to new component named `ProductCardList` component to reuse in multiple places.